nomad/state/state_store.go

```diff
@@ -322,6 +324,7 @@ func (s *StateStore) UpsertJob(index uint64, job *structs.Job) error {
 		job.CreateIndex = index
 		job.ModifyIndex = index
 		job.JobModifyIndex = index
+		job.Version = 0
```

Comment:
Yeah. Probably shouldn't have added the stable field in this PR as I didn't implement it.